Makefile

```
@@ -146,6 +173,129 @@ e2e-helm-install:
 	cd .staging/helm && tar -xvf helmbin.tar.gz
 	./.staging/helm/linux-amd64/helm version --client
 
+e2e-notaryv2-setup:
+	rm -rf .staging/notaryv2
+	mkdir -p .staging/notaryv2
+	cd .staging/notaryv2 && git clone https://github.com/notaryproject/notation
+	cd .staging/notaryv2/notation && make build
+
+	echo 'FROM alpine\nCMD ["echo", "notaryv2 signed image"]' > .staging/notaryv2/Dockerfile
+	docker build -t ${LOCAL_TEST_REGISTRY}/notation:signed .staging/notaryv2
+	docker push ${LOCAL_TEST_REGISTRY}/notation:signed
+
+	docker pull ${LOCAL_UNSIGNED_IMAGE}
+	docker image tag ${LOCAL_UNSIGNED_IMAGE} ${LOCAL_TEST_REGISTRY}/notation:unsigned
+	docker push ${LOCAL_TEST_REGISTRY}/notation:unsigned
+
+	rm -rf ~/.config/notation
+	.staging/notaryv2/notation/bin/notation cert generate-test --default "ratify-bats-test"
+	.staging/notaryv2/notation/bin/notation sign `docker image inspect ${LOCAL_TEST_REGISTRY}/notation:signed | jq -r .[0].RepoDigests[0]`
+	.staging/notaryv2/notation/bin/notation sign `docker image inspect ${LOCAL_TEST_REGISTRY}/all:v0 | jq -r .[0].RepoDigests[0]`  
+
+e2e-cosign-setup:
+	rm -rf .staging/cosign
+	mkdir -p .staging/cosign
+	wget https://github.com/sigstore/cosign/releases/download/v1.13.1/cosign-linux-amd64 
+	mv cosign-linux-amd64 .staging/cosign
+	chmod +x .staging/cosign/cosign-linux-amd64
+
+	echo 'FROM alpine\nCMD ["echo", "cosign signed image"]' > .staging/cosign/Dockerfile
+	docker build -t ${LOCAL_TEST_REGISTRY}/cosign:signed .staging/cosign
+	docker push ${LOCAL_TEST_REGISTRY}/cosign:signed
+
+	docker pull ${LOCAL_UNSIGNED_IMAGE}
+	docker image tag ${LOCAL_UNSIGNED_IMAGE} ${LOCAL_TEST_REGISTRY}/cosign:unsigned
+	docker push ${LOCAL_TEST_REGISTRY}/cosign:unsigned
+
+	export COSIGN_PASSWORD="test" && cd .staging/cosign && ./cosign-linux-amd64 generate-key-pair
+	export COSIGN_PASSWORD="test" && .staging/cosign/cosign-linux-amd64 sign --key .staging/cosign/cosign.key `docker image inspect ${LOCAL_TEST_REGISTRY}/cosign:signed | jq -r .[0].RepoDigests[0]`
+	export COSIGN_PASSWORD="test" && .staging/cosign/cosign-linux-amd64 sign --key .staging/cosign/cosign.key `docker image inspect ${LOCAL_TEST_REGISTRY}/all:v0 | jq -r .[0].RepoDigests[0]`
```

not a strong opinion, maybe this would look better
```suggestion
	export COSIGN_PASSWORD="test" && cd .staging/cosign && \
	./cosign-linux-amd64 generate-key-pair && \
	./cosign-linux-amd64 sign --key .staging/cosign/cosign.key `docker image inspect ${LOCAL_TEST_REGISTRY}/cosign:signed | jq -r .[0].RepoDigests[0]` && \
	./cosign-linux-amd64 sign --key .staging/cosign/cosign.key `docker image inspect ${LOCAL_TEST_REGISTRY}/all:v0 | jq -r .[0].RepoDigests[0]`
```